_search-plugins/search-relevance/rerank-by-field-cross-encoder.md

```diff
@@ -0,0 +1,274 @@
+{ "index": { "_id": 4 } }
+{ "borough": "Manhattan", "area_name": "Harlem", "description": "Harlem is a historic neighborhood in Upper Manhattan, known for its significant African-American cultural heritage.", "population": 116000, "facts": "Harlem was the birthplace of the Harlem Renaissance, a cultural movement that celebrated Black culture through art, music, and literature." } 
+{ "index": { "_id": 5 } }
+{ "borough": "The Bronx", "area_name": "Riverdale", "description": "Riverdale is a suburban-like neighborhood in the Bronx, known for its leafy streets and affluent residential areas.", "population": 48000, "facts": "Riverdale is one of the most affluent areas in the Bronx, with beautiful parks, historic homes, and excellent schools." } 
+{ "index": { "_id": 6 } }
+{ "borough": "Staten Island", "area_name": "St. George", "description": "St. George is the main commercial and cultural center of Staten Island, offering stunning views of Lower Manhattan.", "population": 15000, "facts": "St. George is home to the Staten Island Ferry terminal and is a gateway to Staten Island, offering stunning views of the Statue of Liberty and Ellis Island." }
+```
+{% include copy-curl.html %}
+
+## Step 4: Create a search pipeline
+
+Next, create a search pipeline for reranking. In the search pipeline configuration, the `input_map` and `output_map` define how the input data is prepared for the cross-encoder model and how the model's output is interpreted for reranking:
+
+- The `input_map` specifies which fields in the search documents and the query should be used as inputs to the model:
+    - The `text` field maps to the `facts` field in the indexed documents. It provides the document-specific content that the model will analyze.
+    - The `text_pair` field dynamically retrieves the search query text (`multi_match.query`) from the search request. 
+
+    The combination of `text` (document `facts`) and `text_pair` (search `query`) allows the cross-encoder model to compare the relevance of the document to the query, considering their semantic relationship.
+
+- The `output_map` specifies how the output of the model is mapped to fields in the response:
+    - The `rank_score` field in the response will store the model's relevance score. Both the `_score` and `rank_score` fields will contain the new document score provided by the cross-encoder and will be used to reorder the documents. Because the `rank_score` field will contain duplicate information, the request specifies to remove it from search results. The original score before the reranking is specified to be included in the response for debugging purposes. You can also compare it to the reranked score to see the improvement in search relevancy.
```

Comment:
A little bit confused on this part. It sounds like we are meshing the ml inference processor with the byfield rerank. Ive edited it to talk about them separately
```suggestion
    - The `rank_score` field in the response will store the model's relevance score which will be used to perform reranking.
    
When using the `by_field` rerank type, the `rank_score` field will contain  duplicate information, the request specifies to remove it from search results by asserting `remove_target_field` as true. The original score before the reranking is specified to be included in the response for debugging purposes via `keep_previous_score`. You can also compare it to the reranked score to see the improvement in search relevancy.
```